redash/query_runner/db2.py

```diff
@@ -0,0 +1,147 @@
+import os
+import json
+import logging
+import select
+
+import ibm_db_dbi
+
+from redash.query_runner import *
+from redash.utils import JSONEncoder
+
+logger = logging.getLogger(__name__)
+
+types_map = {
+    ibm_db_dbi.NUMBER: TYPE_INTEGER,
+    ibm_db_dbi.BIGINT: TYPE_INTEGER,
+    ibm_db_dbi.ROWID: TYPE_INTEGER,
+    ibm_db_dbi.FLOAT: TYPE_FLOAT,
+    ibm_db_dbi.DECIMAL: TYPE_FLOAT,
+    ibm_db_dbi.DATE: TYPE_DATE,
+    ibm_db_dbi.TIME: TYPE_DATETIME,
+    ibm_db_dbi.DATETIME: TYPE_DATETIME,
+    ibm_db_dbi.BINARY: TYPE_STRING,
+    ibm_db_dbi.XML: TYPE_STRING,
+    ibm_db_dbi.TEXT: TYPE_STRING,
+    ibm_db_dbi.STRING: TYPE_STRING
+}
```

Comment:
This should be moved into the try/catch block of the import too, as `ibm_db_dbi` might not be available in this case.

Reply:
Ok, rewritten